autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
+        Any valid string path is acceptable.
+    sep : str
+        Delimiter to use.
+    delimiter : str, default ``None``
+        Alias for sep.
+    header : int, list of int, default 'infer'
+        Row number(s) to use as the column names, and the start of the
+        data.
+    names : array-like, optional
+        List of column names to use.
+    index_col : int, str, sequence of int / str, or False, default ``None``
+    usecols : list-like or callable, optional
+        Return a subset of the columns.
+    squeeze : bool, default False
+    prefix : str, optional
```

Comment:
what's that?

Reply:
I stoped here, there a large amount of arguments that I cannot understand, or are they necessary to expose to users? the key idea of autogluon is to have users to specify the minimal arguments. Also, as the general rule, a function should not have too many arguments...

Reply:
Right, I'd remove all these from the docs and just say something end the documentation for this method with something like: 

"Numerous other keyword arguments can be provided which will be passed into pandas.read_csv().  Some you might find most useful include: sep, delimiter, header, etc. [just list <5 or so of the most useful options here]."  

Also as Nick mentioned you can just replace `pd.read_csv()` with `load_pd.load()` used in TabularPrediction, or extend that method if you require more args specific to text datasets (load_pd.load uses pd.read_csv under the hood but brings numerous benefits)

Reply:
This is for documentation. I can improve the documentation if necessary. For load_pd, I think we need detailed documention as well as certain amount of refactory to be shared between tabular and nlp.

Reply:
We should not be documenting all of these options. At most we should maybe document the most critical ones and then have others be optionally specified in some data_loader_args dictionary parameter. You can just provide a link to the pd.read_csv() function which describes the valid parameters. Honestly, we should likely not even provide full functionality and access to all parameters of read_csv. If a user wants to use those functionalities, they can load their data prior to the fit call and pass it in as a DataFrame input to the task. This is how Tabular works.

Reply:
Yes, I think we agree to have both better documentation for nlp data and tabular data now.